File: work_java/day0725/src/test02/Test02.java
package test02;


import test01.Test01;

import java.util.Arrays;
import java.util.Comparator;

public class Test02 {
    public static void main(String[] args) {
        Student[] students = new Student[9];

        students[0] = new Student("조믿음", 25, 95);
        students[1] = new Student("이준호", 22, 85);
        students[2] = new Student("양유진", 20, 100);
        students[3] = new Student("조믿음", 25, 95);
        students[4] = new Student("이준호", 22, 85);
        students[5] = new Student("양유진", 20, 100);
        students[6] = new Student("이준호", 22, 85);
        students[7] = new Student("양유진", 20, 100);
        students[8] = new Student("조믿음", 25, 95);

        Arrays.sort(students, new NameComparator());

        for(Student student : students){
            System.out.println(student);
        }
        System.out.println("------------------------------------");
        Arrays.sort(students, new AgeComparator());

        for(Student student : students){
            System.out.println(student);
        }
    }
}

class AgeComparator implements Comparator<Student>{
    @Override
    public int compare(Student o1, Student o2) {
        return o2.getAge() - o1.getAge();
    }
}

class NameComparator implements Comparator<Student> {
    @Override
    public int compare(Student o1, Student o2) {
        System.out.println("이름비교중");
        return o1.getName().compareTo(o2.getName());
    }
}

class Student {
    private String name;
    private int age;
    private int score;

    public Student(String name, int age, int score) {
        this.name = name;
        this.age = age;
        this.score = score;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Student{" +
                "name='" + name + '\'' +
                ", age=" + age +
                ", score=" + score +
                '}';
    }
}

File: work_kotlin/day1014/src/test01/Test02.kt
package test01

class test02 {
}

class User {
    fun hello() {
        println("객체에 포함되는 함수")
    }
}

fun main() {
    val v1:Int = 100
    val v2 = 200 // 변수에 자료형 생략해도 들어가는 데이터를 보고 유추됨.
    var v3 = 300

    // v1 = 999 // val 은 한번 정하고 나면 변경 불가능
    v3 = 999 // var 은 프로그램 진행하다 값이 변경되어도 괜찮음.

    val user = User() // 객체 생성해서 변수에 기억. 자바 참조개념 그대로. 생성자 호출. new 키워드 생략
    user.hello()

    val java = MyJava("예찬") // 자바로 무조건 클래스에는 생성자 필수로 있었으니, 그 생성자 호출하면서 객체 생성 하면 된다.
}